omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java

```diff
@@ -162,7 +162,7 @@ private boolean isUrlWithinOpenmrs(PageRequest pageRequest, String redirectUrl)
 				try {
 					URL url = new URL(redirectUrl);
 					String urlPath = url.getFile();
-					String urlContextPath = urlPath.substring(0, urlPath.indexOf('/', 1));
+					String urlContextPath = urlPath.substring(1).contains("/") ? urlPath.substring(0, urlPath.indexOf('/', 1)) : urlPath;
```

Comment:
It would be great to add a test for this.

Reply:
Hello @dkayiwa , sure, I just added a new test for this. Thank you.

Reply:
@icrc-jfigueiredo did you notice that your test passes even without the above changes?

Reply:
I did not notice that, but if we try to use something different from "/openmrs" like "/test" on the URL the test will fail.

Reply:
@icrc-jfigueiredo all am trying to say is simply that you are not testing the change. For the test would start by failing and then your changes make it pass.

Reply:
@dkayiwa I updated the test to only pass with my modification, now it really tests the new development.